plugins/modules/cloud/scaleway/scaleway_container_registry.py

```diff
@@ -0,0 +1,265 @@
+    project_id: '{{ scw_project }}'
+    state: present
+    region: fr-par
+    name: my-awesome-container-registry
+  register: container_registry_creation_task
+
+- name: Make sure container regitry is deleted
+  community.general.scaleway_container_registry:
+    project_id: '{{ scw_project }}'
+    state: absent
+    region: fr-par
+    name: my-awesome-container-registry
+'''
+
+RETURN = '''
+container_registry:
+  description: The container registry information.
+  returned: when I(state=present)
+  type: dict
+  sample:
+    created_at: "2022-10-14T09:51:07.949716Z"
+    description: Managed by Ansible
+    endpoint: rg.fr-par.scw.cloud/my-awesome-registry
+    id: 0d7d5270-7864-49c2-920b-9fd6731f3589
+    image_count: 0
+    is_public: false
+    name: my-awesome-registry
+    organization_id: 10697b59-5c34-4d24-8d15-9ff2d3b89f58
+    project_id: 3da4f0b2-06be-4773-8ec4-5dfa435381be
+    region: fr-par
+    size: 0
+    status: ready
+    status_message: ""
+    updated_at: "2022-10-14T09:51:07.949716Z"
+'''
+
+from ansible_collections.community.general.plugins.module_utils.scaleway import (
+    SCALEWAY_ENDPOINT, SCALEWAY_REGIONS, scaleway_argument_spec, Scaleway,
+    scaleway_waitable_resource_argument_spec, resource_attributes_should_be_changed
+)
+from ansible.module_utils.basic import AnsibleModule
+
+STABLE_STATES = (
+    "ready",
+    "absent"
+)
+
+MUTABLE_ATTRIBUTES = (
+    "description",
+    "is_public"
+)
+
+
+def payload_from_wished_cr(wished_cr):
+    playload = {
+        "project_id": wished_cr["project_id"],
+        "name": wished_cr["name"],
+        "description": wished_cr["description"],
+        "is_public": wished_cr["privacy_policy"] == "public"
+    }
+
+    return playload
+
+
+def absent_strategy(api, wished_cr):
+    changed = False
+
+    cr_list = api.fetch_all_resources("namespaces")
+    cr_lookup = dict((cr["name"], cr)
+                     for cr in cr_list)
+
+    if wished_cr["name"] not in cr_lookup.keys():
+        return changed, {}
+
+    target_cr = cr_lookup[wished_cr["name"]]
+    changed = True
+    if api.module.check_mode:
+        return changed, {"status": "Container registry would be destroyed"}
+
+    api.wait_to_complete_state_transition(resource=target_cr, stable_states=STABLE_STATES, force_wait=True)
+    response = api.delete(path=api.api_path + "/%s" % target_cr["id"])
+    if not response.ok:
+        api.module.fail_json(msg='Error deleting container registry [{0}: {1}]'.format(
+            response.status_code, response.json))
+
+    api.wait_to_complete_state_transition(resource=target_cr, stable_states=STABLE_STATES)
+    return changed, response.json
+
+
+def present_strategy(api, wished_cr):
+    changed = False
+
+    cr_list = api.fetch_all_resources("namespaces")
+    cr_lookup = dict((cr["name"], cr)
+                     for cr in cr_list)
+
+    playload_cr = payload_from_wished_cr(wished_cr)
+
+    if wished_cr["name"] not in cr_lookup.keys():
```

Comment:
Simpler and more pythonic as:
```suggestion
    if wished_cr["name"] not in cr_lookup:
```

Reply:
Fixed by 32aad7ae75b8e9ca6888fbb56c61849e1869c88b